api/server/router/network/network_routes.go

```diff
@@ -56,6 +56,10 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 		return err
 	}
 
+	if vars["id"] == "" {
+		return libnetwork.ErrNoSuchNetwork(vars["id"])
```

Comment:
@cpuguy83 yes, see above (https://github.com/docker/docker/pull/24634#issuecomment-232663451), but that would change the response from the current error (if an invalid name is provided); I wasn't sure if we wanted to risk that. Also see https://github.com/docker/docker/pull/24634#issuecomment-232722602